### PR TITLE
added e2e test for oauth flow including signing of contract

### PR DESCRIPTION
**Pull request by woutslakhorst**
It requires some changes on nuts-crypto, nuts-go-core and nuts-auth first.

### CODE REVIEW

File: oauth-flow/run-test.sh [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  echo $RESPONSE | sed -E 's/.*"message":"([^"]*).*/\1/' > ./node-B/data/contract.txt
+  echo "Contract stored in ./node-B/data/contract.txt"
+else
+  echo "FAILED: Could not get contract drawn up at node-B" 1>&2
+  echo $RESPONSE
+  exit 1
+fi
+
+# sign the contract with dummy means
+sed "s/BASE64_CONTRACT/$(cat ./node-B/data/contract.txt)/" ./node-B/createsigningsessionrequesttemplate.json > ./node-B/data/createsigningsessionrequest.json
```
Comment (reinkrul) [inline oauth-flow/run-test.sh:91]: That's some proper Bash-fu you've got going there!

**Review by reinkrul — CHANGES_REQUESTED** [review]
When I try to run using a build from latest master it fails;

```
------------------------------------
Obtain client certificate...
------------------------------------
read EC key
writing EC key
FAILED: Could not get TLS client certificate from node-B
{"message":"Not Found"}
```

What I also see:
```
nodeB_1          | time="2020-12-07T14:43:31Z" level=error msg="Status code: 0" err="*irma.SessionError"
...
nodeB_1          | {"time":"2020-12-07T14:43:34.657283Z","id":"","remote_ip":"172.22.0.1","host":"localhost:21323","method":"POST","uri":"/crypto/certificate/tls","user_agent":"curl/7.64.1","status":404,"error":"code=404, message=Not Found","latency":49200,"latency_human":"49.2µs","bytes_in":178,"bytes_out":24}
```

**Comment by woutslakhorst** [comment]
> When I try to run using a build from latest master it fails;

yes waiting for some other PR's

**Comment by woutslakhorst** [comment]
@reinkrul latest master works now

**Comment by reinkrul** [comment]
> @reinkrul latest master works now

Maste branch of `nuts-go`? I don't see changes there since I last tested, after running it again its still broken?

**Review by reinkrul — APPROVED** [review]
Works, but only on the second time. The first time always leads to a time out while waiting for the services to start.

**Pull request merged**